From 5bfa9f1cd0523a0cad70cacf5ecf8c1aeb06c4d0 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@gmail.com>
Date: Tue, 22 Apr 2025 17:47:02 +0000
Subject: [PATCH] configure: make sure to call AC_CHECK_HEADER_STDBOOL

before using ac_cv_sizeof_bool

It was called after this conditional, so PG_USE_STDBOOL wasn't set
even when it should be as shown in at the end of config.log:

  ac_cv_header_stdbool_h=yes
  ac_cv_sizeof_bool=1
  ac_cv_type__Bool=yes
  #define SIZEOF_BOOL 1
  #define HAVE__BOOL 1
  #define HAVE_STDBOOL_H 1

* fixes:
  https://github.com/postgresql-interfaces/psqlodbc/issues/110
  https://github.com/postgresql-interfaces/psqlodbc/issues/94

Upstream-Status: Submitted [https://github.com/postgresql-interfaces/psqlodbc/pull/112]

Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
 configure.ac | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index f35c825..ea363ca 100644
--- a/configure.ac
+++ b/configure.ac
@@ -28,9 +28,11 @@ AC_CHECK_SIZEOF([bool], [],
 #include <stdbool.h>
 #endif])
 
+AC_CHECK_HEADER_STDBOOL()
+
 dnl We use <stdbool.h> if we have it and it declares type bool as having
 dnl size 1.  Otherwise, c.h will fall back to declaring bool as unsigned char.
-if test "$ac_cv_header_stdbool_h" = yes -a "$ac_cv_sizeof_bool" = 1; then
+if test "$ac_cv_header_stdbool_h" = yes && test "$ac_cv_sizeof_bool" = 1; then
   AC_DEFINE([PG_USE_STDBOOL], 1,
             [Define to 1 to use <stdbool.h> to define type bool.])
 fi
